docs/src/operations/authorization.md

```diff
@@ -32,32 +44,150 @@ public class SecureClient {
 	})
+  if err != nil {
+    panic(err)
+  }
   
-  // continue... 
+  response, err := client.NewTopologyCommand().Send()
+  if err != nil {
+    panic(err)
+  }
+    
+  fmt.Println(response.String())
+}
+```
+## OAuthCredentialsProvider
+
+The OAuthCredentialsProvider requires the specification of a client ID and a client secret. These are then used to request an access token from an OAuth 2.0 authorization server through a [client credentials flow](https://tools.ietf.org/html/rfc6749#section-4.4). By default, the authorization server is the one used by Camunda Cloud but any other can be used. Using the access token returned by the authorization server, the OAuthCredentialsProvider will add it to the gRPC headers of each request as a bearer token. If a requests fails with an UNAUTHENTICATED gRPC code and if the stored access token can be refreshed, the request will be retried. If the failure is caused by any other code or if the refreshed access token is the same as the stale access token, then the request will not be retried.
+
+### Java
+
+To use the a Zeebe client with Camunda Cloud, first an OAuthCredentialsProvider has to be created and configured with the appropriate client credentials. The `audience` is equivalent to the cluster endpoint without a port number.
+
+``` java
+public class AuthorizedClient {
+
+  public void main(String[] args) {
+    final OAuthCredentialsProvider provider =
+        new OAuthCredentialsProviderBuilder()
+            .clientId("clientId")
+            .clientSecret("clientSecret")
+            .audience("cluster.endpoint.com")
+            .build();
+
+    final ZeebeClient client =
+        new ZeebeClientBuilderImpl()
+            .brokerContactPoint("cluster.endpoint.com:443")
+            .credentialsProvider(provider)
+            .build();
+    
+    System.out.println(client.newTopologyRequest().send().join().toString());
+  }
+}
+```
+
+Another way of creating a properly configured client would be through environment variables. After setting the `ZEEBE_CLIENT_ID` and `ZEEBE_CLIENT_SECRET` variables to the correct values, the following would be equivalent to the previous:
+
+```java
+public class AuthorizedClient {
+
+  public void main(String[] args) {
+    final ZeebeClient client =
+        new ZeebeClientBuilderImpl()
+            .brokerContactPoint("cluster.endpoint.com:443")
+            .build();
+
+    System.out.println(client.newTopologyRequest().send().join().toString());
+  }
+}
+```
+
+The client will create a OAuthCredentialProvider with the credentials specified through the environment variables and the audience will be extracted from the the address specified through the ZeebeClientBuilder.
+
+> **Note:** Zeebe's Java client will not prevent you from adding credentials to gRPC calls while using an insecure connection but you should be aware of the risks of doing so. The client will log a warning message whenever such a call is made, to prevent this from happening accidentaly. 
```

Comment:
Two things: `accidentaly` should be `accidentally`, and we should probably mention here what the risks are :sweat_smile: Not sure it's useful to mention we'll log a warning for it though, they'll figure that one out by themselves :wink: